packages/spatial/src/physics/classes/Physics.test.ts

```diff
@@ -361,3 +362,714 @@ describe('Physics', () => {
+        assert.deepEqual(body.linvel(), linVel)
+      })
+
+      it("should set the body's Angular Velocity to the given value", () => {
+        Physics.setRigidbodyPose(testEntity, position, rotation, linVel, angVel)
+        const body = Physics._Rigidbodies.get(testEntity)!
+        assert.deepEqual(body.angvel(), angVel)
+      })
+    })
+
+    describe('enabledCcd', () => {
+      let testEntity = UndefinedEntity
+      let physicsWorld: World | undefined = undefined
+
+      beforeEach(async () => {
+        createEngine()
+        await Physics.load()
+        physicsWorld = Physics.createWorld()
+        getMutableState(PhysicsState).physicsWorld!.set(physicsWorld!)
+        physicsWorld!.timestep = 1 / 60
+
+        // Create the entity
+        testEntity = createEntity()
+        setComponent(testEntity, TransformComponent)
+        setComponent(testEntity, RigidBodyComponent, { type: BodyTypes.Dynamic })
+        Physics.createRigidBody(testEntity, physicsWorld!)
+      })
+
+      afterEach(() => {
+        removeEntity(testEntity)
+        physicsWorld = undefined
+        return destroyEngine()
+      })
+
+      it('should enable Continuous Collision Detection on the entity', () => {
+        const body = Physics._Rigidbodies.get(testEntity)!
+        assert.equal(body.isCcdEnabled(), false)
+        Physics.enabledCcd(testEntity, true)
+        assert.equal(body.isCcdEnabled(), true)
+      })
+
+      it('should disable CCD on the entity when passing `false` to the `enabled` property', () => {
+        const body = Physics._Rigidbodies.get(testEntity)!
+        Physics.enabledCcd(testEntity, true)
+        assert.equal(body.isCcdEnabled(), true)
+        Physics.enabledCcd(testEntity, false)
+        assert.equal(body.isCcdEnabled(), false)
+      })
+    })
+
+    /**
+  // @todo How to check for the impulse of an entity?
+  describe("applyImpulse", () => {
+    let testEntity = UndefinedEntity
+    let physicsWorld :World | undefined= undefined
+
+    beforeEach(async () => {
+      createEngine()
+      await Physics.load()
+      physicsWorld  = Physics.createWorld()
+      getMutableState(PhysicsState).physicsWorld!.set(physicsWorld!)
+      physicsWorld!.timestep = 1 / 60
+
+      // Create the entity
+      testEntity = createEntity()
+      setComponent(testEntity, TransformComponent)
+      setComponent(testEntity, RigidBodyComponent, { type: BodyTypes.Dynamic })
+      Physics.createRigidBody(testEntity, physicsWorld!)
+    })
+
+    afterEach(() => {
+      removeEntity(testEntity)
+      physicsWorld = undefined
+      return destroyEngine()
+    })
+
+    it("should apply the impulse to the RigidBody of the entity", () => {
+      const testImpulse = new Vector3(1,2,3)
+      const body = Physics._Rigidbodies.get(testEntity)!
+      Physics.applyImpulse(testEntity, testImpulse)
+    })
+  })
+  */
+
+    /**
+  // @todo How to check rotations?
+  describe("lockRotations", () => {
```

Comment:
apply an impulse and assert it has not rotated along the locked axis

Reply:
After applying the reference rotation values on creation, the resulting Quat does not remain the same compared to the starting reference. It somehow changes to float values with many decimals that have a lot of variance from run to run due to floating point imprecision.

So, this "how to check rotations" note is for asking about asserting Quaternion/Rotation results, as in how to assert the contained float values themselves.

Reply:
Fixed somewhere in the commit list.   _(forgot to track the commit's sha)_

The mistake was that Quaterions were being normalized internally in Rapier, but I wasn't sending a normalized value.
It was solved by normalizing the input value and the checking the resulting data against that input.